app/controllers/samples_controller.rb

```diff
@@ -16,6 +16,17 @@ def index
     @samples = perform_pagination(@samples)
   end
 
+  def show
+    sample = Sample.find(params[:id])
+    @sample_form = SampleForm.for(sample)
+    return unless authorize_resource(sample, READ_SAMPLE)
```

Comment:
Same here: the template will still be rendered? It will crash because `@view_helper` is undefined.

Reply:
If not authorized the server returns a 403